packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -2,20 +2,165 @@
+  /// ```dart
+  /// _ErrorDiagnostic('Element $element must be $color')
+  /// ```
+  /// Desugars to:
+  /// ```dart
+  /// _ErrorDiagnostic._fromParts(<Object>['Element ', element, ' must be ', color]);
+  /// ```
+  ///
+  /// Slightly more complex case:
+  /// ```dart
+  /// _ErrorDiagnostic('Element ${element.runtimeType} must be $color');
+  /// ```
+  /// Desugars to:
+  ///```dart
+  /// _ErrorDiagnostic._fromParts(<Object>[
+  ///   'Element ',
+  ///   DiagnosticsProperty(null, element, description: element.runtimeType?.toString()),
+  ///   ' must be ',
+  ///   color,
+  /// ]);
+  /// ```
+  /// {@end-tool}
+  _ErrorDiagnostic._fromParts(
+    List<Object> messageParts, {
+    DiagnosticsTreeStyle style = DiagnosticsTreeStyle.flat,
+    DiagnosticLevel level = DiagnosticLevel.info,
+  }) : assert(messageParts != null),
+       super(
+         null,
+         messageParts,
+         showName: false,
+         showSeparator: false,
+         defaultValue: null,
+         style: style,
+         level: level,
+       );
+
+  @override
+  String valueToString({ TextTreeConfiguration parentConfiguration }) {
+    return value.join('');
+  }
+}
+
+/// An explanation of the problem and its cause, any information that may help
+/// track down the problem, background information, etc.
+///
+/// Use [ErrorDescription] for any part of an error message where neither
+/// [ErrorSummary] or [ErrorHint] is appropriate.
+///
+/// See also:
+///
+/// * [ErrorSummary], which provides a short (one line) description of the
+///   problem that was detected.
+/// * [ErrorHint], which provides specific, non-obvious advice that may be
+///   applicable.
+/// * [FlutterError], which is the most common place to use an
+///   [ErrorDescription].
+class ErrorDescription extends _ErrorDiagnostic {
+  /// This constructor provides a reliable hook for a kernel transformer to find
+  /// error messages that need to be rewritten to include object references for
+  /// interactive display of errors.
+  ///
+  /// The message will display with the same text regardless of whether the
+  /// kernel transformer is used. The kernel transformer is required so that
+  /// debugging tools can provide interactive displays of objects described by
+  /// the error.
+  ErrorDescription(String message) : super._fromParts(<Object>[message], level: DiagnosticLevel.info);
+
+  /// Calls to the default constructor may be rewritten to use this constructor
+  /// in debug mode using a kernel transformer.
+  ErrorDescription._fromParts(List<Object> messageParts) : super._fromParts(messageParts, level: DiagnosticLevel.info);
+}
+
+/// A short (one line) description of the problem that was detected.
+///
+/// Error summaries from the same source location should have little variance,
+/// so that they can be recognized as related. For example, they shouldn't
+/// include hash codes.
+///
+/// A [FlutterError] must start with an [ErrorSummary] and may not contain
+/// multiple summaries.
```

Comment:
I think this line needs updating?

Reply:
Sorry can you clarify what about this line should be updated? It is still technically correct that every FlutterError starts with an ErrorSummary although sometimes the ErrorSummary is inferred parsing the text of the message.

Reply:
Sorry, I wasn't clear.  I just wondered if you wanted to remove the reference to the `ErrorSummary` type in favor of a string since that's what folks will be passing in.  A not really so no worries either way!